kyaml/kio/byteio_writer.go

```diff
@@ -55,8 +55,23 @@ func (w ByteWriter) Write(nodes []*yaml.RNode) error {
 		}
 	}
 
-	encoder := yaml.NewEncoder(w.Writer)
-	defer encoder.Close()
+	// Check if the output is a single JSON file so we can force JSON encoding in that case.
+	// YAML flow style encoding may not be compatible because of unquoted strings and newlines
+	// introduced by the YAML marshaller in long string values. These newlines are insignificant
+	// when interpreted as YAML but invalid when interpreted as JSON.
+	jsonEncodeSingleNode := false
```

Comment:
the comment should explain the rationale behind the node slice length == 1 check

So if we have 3 nodes, and the middle one came from a json file, while the other 2 did not,
we don't json encode?  comment should explain

Also, please throw this code and comment into a func, e.g.
```
forceJsonEncoding := shouldForceJsonEncoding(nodes)
```

Reply:
Ok, but I'll name the func and var slightly differently to make it clear why it's fine to assume that the node slice has a single element when special casing further down.